crates/net/downloaders/src/headers/linear.rs

```diff
@@ -873,7 +865,7 @@ impl LinearDownloadBuilder {
     /// and header client implementations
     pub fn build<H>(self, consensus: Arc<dyn Consensus>, client: Arc<H>) -> LinearDownloader<H>
     where
-        H: HeadersClient + 'static,
+        H: HeadersClient<Output = HeadersFut> + 'static,
```

Comment:
sigh, can't believe github just ignored a comment again...

anyway typing this again:

now that we have the associated future we should be able to get rid of the `Output` bound entirely and only rely on `H::Output`

so that for example `HeadersRequestFuture` becomes `HeadersRequestFuture<F: Future...>`

and inside `LinearDownloader` we get:

```rust
  in_progress_queue: FuturesUnordered<<H as HeadersClient>::Output>,
```

instead.

perhaps Output needs to be Unpin, but I'm not quite sure.